src/components/one-time-donation/components/Steps.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react'
+import FirstStep from './FirstStep'
+import SecondStep from './SecondStep'
+import ThirdStep from './ThirdStep'
+import Success from './Success'
+import Unsuccess from './Unsuccess'
+import {
+  DonationBankInput,
+  DonationResponse,
+  DonationStep as StepType,
+  OneTimeDonation,
+} from '../../../gql/donations'
+import { FormikStep, FormikStepper } from './FormikStepper'
+import { useTranslation } from 'next-i18next'
+import { useRouter } from 'next/router'
+import { useMutation, UseQueryResult } from 'react-query'
+import { CampaignResponse } from 'gql/campaigns'
+import { useViewCampaign } from 'common/hooks/campaigns'
+import { useVaultsList } from 'common/hooks/vaults'
+import { useCreateBankDonation } from 'service/donation'
+import { AxiosError, AxiosResponse } from 'axios'
+import { ApiErrors, isAxiosError, matchValidator } from 'service/apiErrors'
+import { FormikHelpers } from 'formik'
+import { validateFirst, validateSecond, validateThird } from '../helpers/validation-schema'
+
+const initialValues: OneTimeDonation = {
+  message: '',
+  anonymous: false,
+  amount: 0,
+  anonymousDonation: true,
+  personsFirstName: '',
+  personsLastName: '',
+  personsEmail: '',
+  personsPhone: '',
+  payment: 'bank',
+}
+
+export default function DonationStepper() {
+  const [status, setStatus] = React.useState(false)
+  const { t } = useTranslation('one-time-donation')
+  const router = useRouter()
+  const slug = String(router.query.slug)
+  const { data }: UseQueryResult<{ campaign: CampaignResponse }> = useViewCampaign(slug as string)
+  const valts = useVaultsList().data
+  const vault = valts?.find((a) => a.campaignId === data?.campaign.id)
+  const mutationFn = useCreateBankDonation()
+
+  const mutation = useMutation<
+    AxiosResponse<DonationResponse>,
+    AxiosError<ApiErrors>,
+    DonationBankInput
+  >({
+    mutationFn,
+    onSuccess: () => {
+      setStatus(true)
+    },
+  })
+  const onSubmit = async (
+    values: OneTimeDonation,
+    { setFieldError, resetForm }: FormikHelpers<OneTimeDonation>,
+  ) => {
+    try {
+      const data = {
+        currency: 'BGN',
+        amount: Number(values.amount),
+        personsEmail: values.personsEmail,
+        personsFirstName: values.personsFirstName,
+        personsLastName: values.personsLastName,
+        personsPhone: values.personsPhone,
+        extCustomerId: String(Math.random() * 5),
+        extPaymentIntentId: String(Math.random() * 5),
+        extPaymentMethodId: String(Math.random() * 5),
+        targetVaultId: vault?.id,
+      }
+      console.log(data)
+      // await mutation.mutateAsync(data)
```

Comment:
This endpoint is only for administrators and regular users will receive `401 Forbidden` when they try to donate

A new endpoint needs to be created which only receives a subset of the params like `campaignId`, without knowledge for the `vaultId`


---


We may leave things here the way they are until we discuss and acknowledge the changes in the backend first.